syntax = "proto3";

message SportOrigin_e_s {
    SportRoot_e_s root = 1;
}

message SportRoot_e_s {
    sport_data_e_s data = 1;
}

message sport_data_e_s {
    sport_data_sport_e_s data_sport = 1;
}

message sport_data_sport_e_s {
    sport_sport_e_s sport = 1;
    int32 number = 3;
    repeated sport_category_events_e_s category = 4;
}

message sport_sport_e_s {
    string sport_id = 1;
    string sport_name = 2;
}

message sport_category_events_e_s {
    sport_events_info_e_s events_info = 1;
}

message sport_events_info_e_s {
    string events_xx = 1;
    string events_category = 2;
    string events_x = 3;
    repeated sport_event_e_s events = 5;
}

message sport_event_e_s {
    int64 event_id = 1;
    string event_name = 2;
    sport_event_time_e_s event_time = 4;
    string number_5 = 5;
    repeated sport_bet_details_e_s bet_details = 6;
    string event_xx = 7;
    int32 number_10 = 10;
    int32 number_12 = 12;
}

message sport_event_time_e_s {
    int64 datetime = 1;
}

message sport_bet_details_e_s {
    int32 bet_category = 1;
    string bet_category_name = 2;
    repeated sport_bet_e_s bet = 3;
}

message sport_bet_e_s {
    string bets_id = 1;
    string bets_name = 2;
    repeated sport_bets_info_e_s bets_info = 6;
    int32 number_11 = 11;
    int32 number_12 = 12;
}

message sport_bets_info_e_s {
    int64 bet_id = 1;
    string bet_name = 2;
    int32 bet_number = 3;
    repeated sport_odd_e_s odds = 4;
}

message sport_odd_e_s {
    string odd_id = 1;
    string odd_name = 2;
    float odd_number = 3;
}